gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ManagedHttpJsonChannel.java

```diff
@@ -82,58 +84,116 @@ public <RequestT, ResponseT> HttpJsonClientCall<RequestT, ResponseT> newCall(
+    return executor;
+  }
+
   @Override
   public synchronized void shutdown() {
+    // Calling shutdown/ shutdownNow() twice should no-op
     if (isTransportShutdown) {
       return;
     }
     try {
+      // Only shutdown the executor if it was created by Gax. External executors
+      // should be managed by the user.
+      if (usingDefaultExecutor && executor instanceof ExecutorService) {
+        ((ExecutorService) executor).shutdown();
+      }
       deadlineScheduledExecutorService.shutdown();
       httpTransport.shutdown();
       isTransportShutdown = true;
     } catch (IOException e) {
-      e.printStackTrace();
+      // TODO: Log this scenario once we implemented the Cloud SDK logging.
+      // Swallow error if httpTransport shutdown fails
     }
   }
 
   @Override
   public boolean isShutdown() {
-    return isTransportShutdown;
+    // TODO(lawrenceqiu): Expose an isShutdown() method for HttpTransport
+    boolean isShutdown = isTransportShutdown && deadlineScheduledExecutorService.isShutdown();
+    // Check that the Gax's ExecutorService is shutdown as well
+    if (usingDefaultExecutor && executor instanceof ExecutorService) {
+      isShutdown = isShutdown && ((ExecutorService) executor).isShutdown();
+    }
+    return isShutdown;
   }
 
   @Override
   public boolean isTerminated() {
-    return isTransportShutdown;
+    boolean isTerminated = deadlineScheduledExecutorService.isTerminated();
+    // Check that the Gax's ExecutorService is terminated as well
+    if (usingDefaultExecutor && executor instanceof ExecutorService) {
+      isTerminated = isTerminated && ((ExecutorService) executor).isTerminated();
+    }
+    return isTerminated;
   }
 
   @Override
   public void shutdownNow() {
-    shutdown();
+    // Calling shutdown/ shutdownNow() twice should no-op
+    if (isTransportShutdown) {
+      return;
+    }
+    try {
+      // Only shutdown the executor if it was created by Gax. External executors
+      // should be managed by the user.
+      if (usingDefaultExecutor && executor instanceof ExecutorService) {
+        ((ExecutorService) executor).shutdownNow();
+      }
+      deadlineScheduledExecutorService.shutdownNow();
+      httpTransport.shutdown();
+      isTransportShutdown = true;
+    } catch (IOException e) {
+      // TODO: Log this scenario once we implemented the Cloud SDK logging.
+      // Swallow error if httpTransport shutdown fails
+    }
   }
 
   @Override
   public boolean awaitTermination(long duration, TimeUnit unit) throws InterruptedException {
-    // TODO
-    return false;
+    long endTimeNanos = System.nanoTime() + unit.toNanos(duration);
+    long awaitTimeNanos = endTimeNanos - System.nanoTime();
+    if (awaitTimeNanos <= 0) {
+      return false;
+    }
+    // Only awaitTermination for the executor if it was created by Gax. External executors
+    // should be managed by the user.
+    if (usingDefaultExecutor && executor instanceof ExecutorService) {
```

Comment:
This `usingDefaultExecutor && executor instanceof ExecutorService` is used 4 times and can be extracted to a private method like `shouldManageExecutor()`.